pyquil/quilatom.py

```diff
@@ -537,6 +540,37 @@ def __neg__(self) -> "Mul":
     def _substitute(self, d: Any) -> ExpressionDesignator:
         return self
 
+    def _simplify(self) -> quil_rs_expr.Expression:
+        rs_expression = _convert_to_rs_expression(self)
+        rs_expression.simplify()
+        return rs_expression
+
+    def __int__(self) -> int:
+        return int(float(self))
+
+    def __float__(self) -> float:
+        expr = self._simplify()
+        if expr.is_number():
+            n = expr.to_number()
+            if n.imag == 0:
+                return float(n.real)
+            else:
+                raise TypeError(f"Expression {self} simplifies to a complex number with a non-zero imaginary part, can't cast to a float.")
+        if expr.is_pi():
+            return np.pi
```

Comment:
Worth noting, `quil-rs` should do this itself, but I can see wanting the extra assurance here